clients/venice-push-job/src/test/java/com/linkedin/venice/hadoop/VenicePushJobTest.java

```diff
@@ -682,6 +684,14 @@ public void testTargetedRegionPushPostValidationFailedForValidation() throws Exc
     verify(pushJob, never()).postValidationConsumption(any());
   }
 
+  @Test(expectedExceptions = VeniceException.class, expectedExceptionsMessageRegExp = "D2 zk host and service name must be provided when schema reader is enabled")
```

Comment:
Is it possible to write an integration test for this?

I know it is hard, and if not, let us try to write more comprehensive unit tests since the current unit test seems to be dummy.

Reply:
Sure. After taking another look, I find that adding an integ test for this is not as complex as I thought. There is an existing test class `ConsumerIntegrationTest` I can leverage. This class can create a venice writer which produces records with new KME into a kafka topic.

Therefore, I added `testKIFRepushForwardCompatibility` which produces records with new KME into VT and then runs repush job. The test passes when schema reader is enabled and fails otherwise with exception:
```
java.lang.Exception: com.linkedin.venice.exceptions.VeniceMessageException: Received Protocol Version '12' which is not supported by OptimizedKafkaValueSerializer. Protocol forward compatibility is not enabled. The only supported Protocol Versions are: {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}.
```